### ISSUE TITLE
WKTWriter doesn't create the expected result

### ISSUE THREAD

**Issue by jany-tenaj** [NetTopologySuite/NetTopologySuite#156]
I've been trying to create WKT strings from Linestrings with the help of WKTWriter and got the following unexpected results. It would be nice if someone could tell me whether WKTWriter was supposed to work this way.

1. The comment from WKTWriter.Write indicates that the function returns a Geometry Tagged Text string  according to the [OpenGIS Simple Features Specification](http://www.opengeospatial.org/standards/sfa). Therefore I would expect that the following code would return "LINESTRING Z (1 2 7, 3 4 5, 5 6 3, 7 8 9, 1 2 7)". 
But the current result "LINESTRING (1 2 7, 3 4 5, 5 6 3, 7 8 9, 1 2 7)" is missing the Z. This problem exists for M and MZ geometries also.

```
            double[] xy = { 1, 2, 3, 4, 5, 6, 7, 8, 1, 2 };
            double[] z = { 7, 5, 3, 9, 7 };
            var cs = new DotSpatialAffineCoordinateSequence(xy, z);
            var ls = new LineString(cs, GeometryFactory.Default);

            WKTWriter wkt = new WKTWriter(3);
            var res2 = wkt.Write(ls);
```

2. EmitM and AllowedOrdinates indicate that WKTWriter would be able to write wkt-strings with M value. Currently there is no way to write a XYMZ WKT as 4 dimensions are not allowed. But even the following code that tries to create a 3 dimensional WKT with M values by omitting Z values doesn't create XYM WKTs.
```
            double[] xy = { 1, 2, 3, 4, 5, 6, 7, 8, 1, 2 };
            double[] z = { 7, 5, 3, 9, 7 };
            double[] m = { 1, 2, 3, 6, 4 };
            var cs = new DotSpatialAffineCoordinateSequence(xy, z, m);
            var ls = new LineString(cs, GeometryFactory.Default);

            WKTWriter wkt = new WKTWriter(3);
            wkt.EmitM = true;
            wkt.EmitZ = false;
            var res2 = wkt.Write(ls);
```

3. EmitZ, EmitM are ignored. The following code writes XYZ WKTs although EmitZ is set to false.
```
            double[] xy = { 1, 2, 3, 4, 5, 6, 7, 8, 1, 2 };
            double[] z = { 7, 5, 3, 9, 7 };
            var cs = new DotSpatialAffineCoordinateSequence(xy, z);
            var ls = new LineString(cs, GeometryFactory.Default);

            WKTWriter wkt = new WKTWriter(3);
            wkt.EmitZ = false;
            var res2 = wkt.Write(ls);
```

**Comment by airbreather** [comment]
Ideally, whatever we do to fix this should be readable once we do #190, correct?

**Comment by airbreather** [comment]
TODO: re-evaluate this after porting the JTS 1.16 changes.

**Comment by airbreather** [comment]
As of 91cd26e, here's the status on the three items in this report:

1. We now emit "`LINESTRING Z`" / "`LINESTRING M`" / "`LINESTRING ZM`" instead of just "`LINESTRING`".
1. `EmitSRID`, `EmitZ`, `EmitM`, `HandleSRID`, and `HandleOrdinates` still do nothing to control the output, so you still have to make sure the geometry contains only what you want us to write out.  However, we do support writing ZM geometries now.
1. See previous item: all those flags are still ignored.

**Comment by airbreather** [comment]
Resolved in f4b3d6a.  Output is controlled by setting `OutputOrdinates`.

`EmitSRID` wasn't mentioned here, but it's gone too.  It may come back, but only after we implement it to actually do something.
